projects/libyuv/libyuv_fuzzer.cc

```diff
@@ -0,0 +1,78 @@
+/* # Copyright 2020 Google Inc. */
+/* # */
+/* # Licensed under the Apache License, Version 2.0 (the "License"); */
+/* # you may not use this file except in compliance with the License. */
+/* # You may obtain a copy of the License at */
+/* # */
+/* #      http://www.apache.org/licenses/LICENSE-2.0 */
+/* # */
+/* # Unless required by applicable law or agreed to in writing, software */
+/* # distributed under the License is distributed on an "AS IS" BASIS, */
+/* # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. */
+/* # See the License for the specific language governing permissions and */
+/* # limitations under the License. */
+/* # */
+/* ################################################################################ */
+
+#include <iostream>
+#include <vector>
+#include <climits>
+
+#include <fuzzer/FuzzedDataProvider.h>
+#include "libyuv.h"
+#include "fuzz_common.h"
+
+using namespace std;
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+    if (size == 0)
+    {
+        libyuv::I420ToNV21(data, 0, data, 0, data, 0, nullptr, 0, nullptr, 0, 0, 0);
+        return 0;
+    }
+
+    FuzzedDataProvider fuzzed_data(data, size);
+
+    int range_size = (static_cast<int>(size) + 1) / 2;
+    int width = 0, height = 0, kPixels = 0, kHalfPixels = 0;
+    do
+    {
+        height = fuzzed_data.ConsumeIntegralInRange(0, range_size);
+        width = fuzzed_data.ConsumeIntegralInRange(0, range_size) ;
+        kPixels = width * height;
+        kHalfPixels = (width + 1) / 2 * (height + 1) / 2;
+    }
+    while (kPixels + kHalfPixels * 2 > size);
```

Comment:
This loop effectively doesnt test odd widths, since they round the size up.  It wont crash but it wont test interesting values

Reply:
The 44th line should be **kHalfPixels=(( width + 1) / 2 )*(( height + 1 ) / 2 )**, I forgot to modify it.
I am sorry that I can't understand your meaning. ”widths“ is generated randomly by the fuzzer, and sometimes is odd. so what do "widths" mean?  Could you tell me in more detail?  just give me an example. thanks!

Reply:
Please give me some more details. I am always waiting for your news. Thanks!

Reply:
@fbarchard I not sure whether you are recently busy or not, but if I disturbed you, I am sorry. I look forward to hearing from you about more details on the issue soon.

Reply:
I'm not the best person to review fuzz targets.  An owner of the oss-fuzz should approve if they have someone willing to take on the support for the fuzz target itself.
I think you've got the basic call to the I420ToNV21 function right, but I think the fuzz end of things is wrong.
Someone familiar with fuzz should review.  There seems to be 2 issues
1. you are given a 'size' in bytes, and from that you attempt to come up with a width and height that fit.  But if they dont, you loop.  It would be better to not have a loop, but compute a width and height that are expected to work.
For example come up with a width that is random and less than size or sqrt of size, and then calculate a height that will make an image less than size bytes.

2. you allocate 'size' bytes for destination.
thats not strictly right.  the source and destination formats are different.
In general it should compute how many bytes the destination will require based on the width and height.
I assume in fuzz we dont care if the destination size is larger than 'size' which is used for source.  It should allocate a buffer that is large enough.